crates/lsp/src/lib.rs

```diff
@@ -167,57 +130,52 @@ fn on_shutdown(
 }
 
 fn on_code_lens_request(
-    state: &mut LspState,
+    _state: &mut LspState,
     params: CodeLensParams,
 ) -> impl Future<Output = Result<Option<Vec<CodeLens>>, ResponseError>> {
-    let file_path = &params.text_document.uri.to_file_path().unwrap();
+    let file_path = params.text_document.uri.to_file_path().unwrap();
+    let program_dir = file_path.parent().unwrap();
 
-    let mut context = match &state.root_path {
-        Some(root_path) => {
-            let fm = FileManager::new(root_path);
-            let graph = CrateGraph::default();
-            Context::new(fm, graph)
-        }
-        None => {
-            let err = ResponseError::new(
-                ErrorCode::REQUEST_FAILED,
-                "Unable to determine the project root path",
-            );
-            return future::ready(Err(err));
-        }
-    };
+    let toml_path = find_package_manifest(program_dir).unwrap();
```

Comment:
Nit. We can pass `file_path` directly into `find_package_manifest` here.

Reply:
I'd like to keep it the program dir, but that reminds me that these unwraps need to be handled gracefully 🤦

Reply:
I suppose the argument is called `current_path` so that should work. I'll make the change.